### PR TITLE
Implement charmstore.Resolve.

### PR DESCRIPTION
**Pull request by frankban**
It can be used to resolve charm series.

### CODE REVIEW

**Comment by mhilton** [comment]
:+1: LGTM

File: charmrepo/charmstore_test.go [hunk 1]
```diff
@@ -190,6 +190,59 @@ func (s *charmStoreRepoSuite) TestLatest(c *gc.C) {
...
+		url: "cs:~who/precise/wordpress",
+	}, {
+		id:  "~who/wordpress-1",
+		url: "cs:~who/precise/wordpress-1",
+	}, {
+		id:  "~dalek/riak",
+		url: "cs:~dalek/utopic/riak",
+	}, {
+		id:  "~dalek/utopic/riak-2",
+		url: "cs:~dalek/utopic/riak-2",
```
Comment (urosj) [inline charmrepo/charmstore_test.go]: Not that it matters, but should this be -3 as -2 holds "wordpress" content?

**Comment by urosj** [comment]
:+1: Ty.

File: charmrepo/charmstore_test.go (hunk 1 above)
Comment (frankban) [inline charmrepo/charmstore_test.go]: It holds wordpress content but that content is still associated with the riak reference.

**Pull request merged**